src/wrover_status_light/ServiceIntro.md

```diff
@@ -0,0 +1,160 @@
+* Server node: the server node receives and handles the service request calls from client nodes. It retrieves data from the request message, performs the necessary actions, and generates a response message which is sent back to the client. 
+* Request and response messages: these messages are structured data containers that are defined by the service definition. The client and server use these messages to exchange information.
+
+## (Optional) Why do we use Services when there's already Topics?
+
+While both services and topics work to allow communication between nodes, they differ in terms of their data exchange patterns and the types of interactions they support. 
+
+Unlike publishers and subscribers over topics, which operate asynchronously, service calls provide synchronous communication. The client node waits for the server node to process the request and receive a response before continuing its execution. This synchronous behavior is beneficial in cases where a response is required before proceeding with the next steps of a program.
+
+In addition, topics are designed for data streaming and broadcasting continuous streams of information, such as sensor data or robot state updates. They are ideal when nodes require a constant flow of information and do not require immediate responses. On the other hand, services are typically used for accessing functionality that is provided by another node. They are well-suited for situations where nodes need to invoke specific actions or request information from another node, and a response is expected.
+
+## Defining a Service
+Services are defined by srv files, which are identical to msg files in terms of formatting, and should be located in the `/srv` folder in a package. In this example, the service takes in a `string` and an `int` as input and returns a `float` as output. 
+```
+#request fields
+string input1
+int32 input2
+---
+#response fields
+float32 output
+```
+After defining a message, ROS would convert the srv file into three classes: service definition, request messages, and response messages. Suppose we have `ExampleService.srv` in an package called `ExamplePackage`, then the names of these packages would be:
+* Service definition: `ExmaplePackage.srv.ExampleService` for Python and `ExamplePackage::ExampleService` for C++
+* Request messages: `ExamplePackage.srv.ExampleServiceRequest` for Python and `ExmaplePackage::ExampleService::Request` for C++
+* Response messages: `ExamplePackage.srv.ExampleServiceResponse` for Python and `ExamplePackage::ExampleService::Request` for C++
+
+## Writing a Server Node
+### Python
+A server code can be initailized with
+```
+def example_service_server():
+    rospy.init_node('example_service_server')
+    s = rospy.Service('example_service', ExamplePackage.srv.ExampleService, example_service_execution)
+    rospy.spin()
+
+def example_service_execution(reqest):
```

Comment:
Using type annotations on the Python code may make it more explicit what one could do with each variable.
```suggestion
def example_service_execution(reqest: ExampleServiceRequest) -> ExampleServiceResponse:
```